controllers/users_config_map_controller.go

```diff
@@ -0,0 +1,281 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	imagerepositoryv1alpha1 "github.com/konflux-ci/image-controller/api/v1alpha1"
+	l "github.com/konflux-ci/image-controller/pkg/logs"
+	"github.com/konflux-ci/image-controller/pkg/quay"
+)
+
+const (
+	ConfigMapFinalizer = "appstudio.openshift.io/quay-team-users"
+)
+
+// QuayUsersConfigMapReconciler reconciles a ConfigMap object with users
+type QuayUsersConfigMapReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	QuayClient       quay.QuayService
+	BuildQuayClient  func(logr.Logger) quay.QuayService
+	QuayOrganization string
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *QuayUsersConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}, builder.WithPredicates(predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				new, ok := e.Object.(*corev1.ConfigMap)
+				if !ok {
+					return false
+				}
+				return IsAdditionalUsersConfigMap(new)
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				new, ok := e.ObjectNew.(*corev1.ConfigMap)
+				if !ok {
+					return false
+				}
+				return IsAdditionalUsersConfigMap(new)
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				new, ok := e.Object.(*corev1.ConfigMap)
+				if !ok {
+					return false
+				}
+				return IsAdditionalUsersConfigMap(new)
+			},
+			GenericFunc: func(e event.GenericEvent) bool {
+				new, ok := e.Object.(*corev1.ConfigMap)
+				if !ok {
+					return false
+				}
+				return IsAdditionalUsersConfigMap(new)
+			},
+		})).
+		Complete(r)
+}
+
+func IsAdditionalUsersConfigMap(configMap *corev1.ConfigMap) bool {
+	return configMap.Name == additionalUsersConfigMapName
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=imagerepositories,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;update
+
+func (r *QuayUsersConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrllog.FromContext(ctx).WithName("ConfigMap")
```

Comment:
Just `ConfigMap` in logs is too generic (logs from all controllers will be mixed).

Reply:
renamed to "QuayUsersConfigMap"